packages/rs-sdk/src/platform/fetch_unproved.rs

```diff
@@ -0,0 +1,75 @@
+use crate::error::Error;
+use crate::platform::proto;
+use crate::Sdk;
+use dapi_grpc::platform::v0::get_current_quorums_info_request::GetCurrentQuorumsInfoRequestV0;
+use dapi_grpc::platform::v0::{self as platform_proto};
+use drive_proof_verifier::unproved::FromUnproved;
+use rs_dapi_client::{transport::TransportRequest, DapiRequest, RequestSettings};
+use std::fmt::Debug;
+
+#[async_trait::async_trait]
+pub trait FetchUnproved
+where
+    Self: Sized + Debug,
+{
+    /// Type of request used to fetch data from Platform.
+    type Request: TransportRequest;
+
+    /// Fetch unproved data from the Platform.
+    ///
+    /// ## Parameters
+    /// - `sdk`: An instance of [Sdk].
+    ///
+    /// ## Returns
+    /// Returns:
+    /// * `Ok(Some(Self))` when object is found.
+    /// * `Ok(None)` when object is not found.
+    /// * [`Err(Error)`](Error) when an error occurs.
+    async fn fetch_unproved(sdk: &Sdk) -> Result<Option<Self>, Error> {
+        Self::fetch_unproved_with_settings(sdk, RequestSettings::default()).await
+    }
+
+    /// Fetch unproved data from the Platform with custom settings.
+    ///
+    /// ## Parameters
+    /// - `sdk`: An instance of [Sdk].
+    /// - `settings`: Request settings for the connection to Platform.
+    ///
+    /// ## Returns
+    /// Returns:
+    /// * `Ok(Some(Self))` when object is found.
+    /// * `Ok(None)` when object is not found.
+    /// * [`Err(Error)`](Error) when an error occurs.
+    async fn fetch_unproved_with_settings(
```

Comment:
Can we have a default implementation here, in a similar way to Fetch?

Reply:
Let's just do that in another PR.